// allure-js-commons/dist/cjs/sdk/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "extractMetadataFromString", {
  enumerable: true,
  get: function get() {
    return _utils.extractMetadataFromString;
  }
});
Object.defineProperty(exports, "getMessageAndTraceFromError", {
  enumerable: true,
  get: function get() {
    return _utils.getMessageAndTraceFromError;
  }
});
Object.defineProperty(exports, "getStatusFromError", {
  enumerable: true,
  get: function get() {
    return _utils.getStatusFromError;
  }
});
Object.defineProperty(exports, "getUnfinishedStepsMessages", {
  enumerable: true,
  get: function get() {
    return _utils.getUnfinishedStepsMessages;
  }
});
Object.defineProperty(exports, "hasLabel", {
  enumerable: true,
  get: function get() {
    return _utils.hasLabel;
  }
});
Object.defineProperty(exports, "hasStepMessage", {
  enumerable: true,
  get: function get() {
    return _utils.hasStepMessage;
  }
});
Object.defineProperty(exports, "isAllStepsEnded", {
  enumerable: true,
  get: function get() {
    return _utils.isAllStepsEnded;
  }
});
Object.defineProperty(exports, "isAnyStepFailed", {
  enumerable: true,
  get: function get() {
    return _utils.isAnyStepFailed;
  }
});
Object.defineProperty(exports, "isMetadataTag", {
  enumerable: true,
  get: function get() {
    return _utils.isMetadataTag;
  }
});
Object.defineProperty(exports, "isPromise", {
  enumerable: true,
  get: function get() {
    return _utils.isPromise;
  }
});
Object.defineProperty(exports, "stripAnsi", {
  enumerable: true,
  get: function get() {
    return _utils.stripAnsi;
  }
});
var _utils = require("./utils.js");
