// allure-js-commons/dist/cjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ContentType", {
  enumerable: true,
  get: function get() {
    return _model.ContentType;
  }
});
Object.defineProperty(exports, "LabelName", {
  enumerable: true,
  get: function get() {
    return _model.LabelName;
  }
});
Object.defineProperty(exports, "LinkType", {
  enumerable: true,
  get: function get() {
    return _model.LinkType;
  }
});
Object.defineProperty(exports, "Severity", {
  enumerable: true,
  get: function get() {
    return _model.Severity;
  }
});
Object.defineProperty(exports, "Stage", {
  enumerable: true,
  get: function get() {
    return _model.Stage;
  }
});
Object.defineProperty(exports, "Status", {
  enumerable: true,
  get: function get() {
    return _model.Status;
  }
});
Object.defineProperty(exports, "StatusByPriority", {
  enumerable: true,
  get: function get() {
    return _model.StatusByPriority;
  }
});
Object.defineProperty(exports, "allureId", {
  enumerable: true,
  get: function get() {
    return _facade.allureId;
  }
});
Object.defineProperty(exports, "attachment", {
  enumerable: true,
  get: function get() {
    return _facade.attachment;
  }
});
Object.defineProperty(exports, "attachmentPath", {
  enumerable: true,
  get: function get() {
    return _facade.attachmentPath;
  }
});
Object.defineProperty(exports, "description", {
  enumerable: true,
  get: function get() {
    return _facade.description;
  }
});
Object.defineProperty(exports, "descriptionHtml", {
  enumerable: true,
  get: function get() {
    return _facade.descriptionHtml;
  }
});
Object.defineProperty(exports, "displayName", {
  enumerable: true,
  get: function get() {
    return _facade.displayName;
  }
});
Object.defineProperty(exports, "epic", {
  enumerable: true,
  get: function get() {
    return _facade.epic;
  }
});
Object.defineProperty(exports, "feature", {
  enumerable: true,
  get: function get() {
    return _facade.feature;
  }
});
Object.defineProperty(exports, "historyId", {
  enumerable: true,
  get: function get() {
    return _facade.historyId;
  }
});
Object.defineProperty(exports, "issue", {
  enumerable: true,
  get: function get() {
    return _facade.issue;
  }
});
Object.defineProperty(exports, "label", {
  enumerable: true,
  get: function get() {
    return _facade.label;
  }
});
Object.defineProperty(exports, "labels", {
  enumerable: true,
  get: function get() {
    return _facade.labels;
  }
});
Object.defineProperty(exports, "layer", {
  enumerable: true,
  get: function get() {
    return _facade.layer;
  }
});
Object.defineProperty(exports, "link", {
  enumerable: true,
  get: function get() {
    return _facade.link;
  }
});
Object.defineProperty(exports, "links", {
  enumerable: true,
  get: function get() {
    return _facade.links;
  }
});
Object.defineProperty(exports, "logStep", {
  enumerable: true,
  get: function get() {
    return _facade.logStep;
  }
});
Object.defineProperty(exports, "owner", {
  enumerable: true,
  get: function get() {
    return _facade.owner;
  }
});
Object.defineProperty(exports, "parameter", {
  enumerable: true,
  get: function get() {
    return _facade.parameter;
  }
});
Object.defineProperty(exports, "parentSuite", {
  enumerable: true,
  get: function get() {
    return _facade.parentSuite;
  }
});
Object.defineProperty(exports, "severity", {
  enumerable: true,
  get: function get() {
    return _facade.severity;
  }
});
Object.defineProperty(exports, "step", {
  enumerable: true,
  get: function get() {
    return _facade.step;
  }
});
Object.defineProperty(exports, "story", {
  enumerable: true,
  get: function get() {
    return _facade.story;
  }
});
Object.defineProperty(exports, "subSuite", {
  enumerable: true,
  get: function get() {
    return _facade.subSuite;
  }
});
Object.defineProperty(exports, "suite", {
  enumerable: true,
  get: function get() {
    return _facade.suite;
  }
});
Object.defineProperty(exports, "tag", {
  enumerable: true,
  get: function get() {
    return _facade.tag;
  }
});
Object.defineProperty(exports, "tags", {
  enumerable: true,
  get: function get() {
    return _facade.tags;
  }
});
Object.defineProperty(exports, "testCaseId", {
  enumerable: true,
  get: function get() {
    return _facade.testCaseId;
  }
});
Object.defineProperty(exports, "tms", {
  enumerable: true,
  get: function get() {
    return _facade.tms;
  }
});
var _facade = require("./facade.js");
var _model = require("./model.js");
