// allure-js-commons/dist/cjs/sdk/reporter/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  ALLURE_METADATA_CONTENT_TYPE: true,
  ALLURE_RUNTIME_MESSAGE_CONTENT_TYPE: true,
  LifecycleState: true,
  ReporterRuntime: true,
  InMemoryWriter: true,
  FileSystemWriter: true,
  MessageWriter: true,
  MessageReader: true,
  getEnvironmentLabels: true,
  getHostLabel: true,
  getThreadLabel: true
};
Object.defineProperty(exports, "ALLURE_METADATA_CONTENT_TYPE", {
  enumerable: true,
  get: function get() {
    return _types.ALLURE_METADATA_CONTENT_TYPE;
  }
});
Object.defineProperty(exports, "ALLURE_RUNTIME_MESSAGE_CONTENT_TYPE", {
  enumerable: true,
  get: function get() {
    return _types.ALLURE_RUNTIME_MESSAGE_CONTENT_TYPE;
  }
});
Object.defineProperty(exports, "FileSystemWriter", {
  enumerable: true,
  get: function get() {
    return _FileSystemWriter.FileSystemWriter;
  }
});
Object.defineProperty(exports, "InMemoryWriter", {
  enumerable: true,
  get: function get() {
    return _InMemoryWriter.InMemoryWriter;
  }
});
Object.defineProperty(exports, "LifecycleState", {
  enumerable: true,
  get: function get() {
    return _LifecycleState.LifecycleState;
  }
});
Object.defineProperty(exports, "MessageReader", {
  enumerable: true,
  get: function get() {
    return _MessageReader.MessageReader;
  }
});
Object.defineProperty(exports, "MessageWriter", {
  enumerable: true,
  get: function get() {
    return _MessageWriter.MessageWriter;
  }
});
Object.defineProperty(exports, "ReporterRuntime", {
  enumerable: true,
  get: function get() {
    return _ReporterRuntime.ReporterRuntime;
  }
});
Object.defineProperty(exports, "getEnvironmentLabels", {
  enumerable: true,
  get: function get() {
    return _labels.getEnvironmentLabels;
  }
});
Object.defineProperty(exports, "getHostLabel", {
  enumerable: true,
  get: function get() {
    return _labels.getHostLabel;
  }
});
Object.defineProperty(exports, "getThreadLabel", {
  enumerable: true,
  get: function get() {
    return _labels.getThreadLabel;
  }
});
var _types = require("./types.js");
var _utils = require("./utils.js");
Object.keys(_utils).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _utils[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _utils[key];
    }
  });
});
var _testplan = require("./testplan.js");
Object.keys(_testplan).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _testplan[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _testplan[key];
    }
  });
});
var _factory = require("./factory.js");
Object.keys(_factory).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _factory[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _factory[key];
    }
  });
});
var _LifecycleState = require("./LifecycleState.js");
var _ReporterRuntime = require("./ReporterRuntime.js");
var _InMemoryWriter = require("./writer/InMemoryWriter.js");
var _FileSystemWriter = require("./writer/FileSystemWriter.js");
var _MessageWriter = require("./writer/MessageWriter.js");
var _MessageReader = require("./writer/MessageReader.js");
var _labels = require("./utils/labels.js");
